### PR TITLE
Initial code drop

### PR DESCRIPTION
**Pull request by brianjmurrell**
Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

### CODE REVIEW

**Comment by jgrund** [comment]
Might be useful to add a Readme explaining what this module does at a high level.

**Comment by brianjmurrell** [comment]
> Might be useful to add a Readme explaining what this module does at a high level.

Would the commit comment dropped into the [READ.md](https://github.com/intel-hpdd/module-tools/blob/master/README.md) be useful?

File: git-versioning.mk [hunk 1]
```diff
@@ -0,0 +1,57 @@
+space :=
+space +=
+SCM_COMMIT_NUMBER	:= $(shell git rev-list HEAD | wc -l)
+ifeq ($(strip $(JOB_NAME)),)
+JENKINS_BUILD_TAG	:=
+else
+JENKINS_BUILD_TAG	:= $(shell echo .jenkins-$(JOB_NAME)-$(BUILD_NUMBER) | \
```
Comment (jgrund) [inline git-versioning.mk]: How does / will this work without Jenkins?

File: rpm-common.mk [hunk 2]
```diff
@@ -0,0 +1,129 @@
+RPM_SPEC      := python-$(NAME).spec
```
Comment (jgrund) [inline rpm-common.mk]: Should this be renamed `python-rpm-common`?

**Comment by jgrund** [comment]
Yes, but I'm thinking along the lines of what this module is meant for and at a high level what you can do with the make targets.

**Comment by jgrund** [comment]
I'm wondering if we can make this CI / RPM host agnostic, I have a workflow going using travis / bintray that seems to be working at the moment

File: copr.mk [hunk 3]
```diff
@@ -0,0 +1,37 @@
+ifneq ($(filter iml_%,$(MAKECMDGOALS)),)
```
[the remark this answers is not in the document]
Comment (brianjmurrell) [reply copr.mk]: It's the targets that were given to `make`.  So in the command `make foo bar`, `MAKECMDGOALS` would be `foo bar`.

It basically let's a `Makefile` know what the user asked to be made.

File: git-versioning.mk (hunk 1 above)
Comment (brianjmurrell) [reply git-versioning.mk]: Outside of Jenkins, `JENKINS_BUILD_TAG` is empty which leads to a package NVR built without any Jenkins build job and number information in it.  I.e.:

`python2-iml-common1.3-1.3.3-27.3.g8eef474.el7.noarch.rpm` vs. `python2-iml-common1.3-1.3.3-27.3.g8eef474.jenkins_iml_common_18.el7.noarch.rpm`.

**Comment by brianjmurrell** [comment]
> I'm wondering if we can make this CI / RPM host agnostic, I have a workflow going using travis / bintray that seems to be working at the moment.

I think it already is for the most part.  It doesn't **depend** on being run in Jenkins for example, it just adds more useful information if it is.  I use this module to build the other modules on my local builder here completely outside of Jenkins and it builds sane targets.

File: rpm-common.mk (hunk 2 above)
Comment (brianjmurrell) [reply rpm-common.mk]: In this context `$(NAME)` is the name of the enclosing module, so *iml-common*, or *iml-agent*, as examples.

File: git-versioning.mk [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	# The commits-since-tag number will ensure that newer packages
+	# are preferred, since RPM's version parsing works left-to-right.
+	PACKAGE_RELEASE := $(BUILD_NUMBER).$(PACKAGE_RELEASE)$(JENKINS_BUILD_TAG)
+
+	# Display this in the UI to make ID easier in dev/test
+	BUILD_NUMBER := $(JENKINS_BUILD_TAG)
+endif
+
+$(shell { echo 'VERSION = "$(VERSION)"';                                    \
+	  echo 'PACKAGE_VERSION = "$(PACKAGE_VERSION)"';                    \
```
Comment (tanabarr) [inline git-versioning.mk]: block should be indented by one

File: include [hunk 5]
```diff
@@ -0,0 +1 @@
+.
```
Comment (tanabarr) [inline include]: what is this doing?

File: git-versioning.mk (hunk 4 above)
Comment (brianjmurrell) [reply git-versioning.mk]: This must be a visualization/rendering problem on GitHub.  In the actual file, the `echo` on line 51 (and all of the lines below it) lines up under the `echo` on line 50.

File: include (hunk 5 above)
Comment (brianjmurrell) [reply include]: It's a symlink so that the module-tools module works in the module-tools repo.

**Pull request merged**